Assets/UI/worldinput_CQUI.lua

```diff
@@ -0,0 +1,559 @@
+-- CQUI Base Extension Functions
+-- Each of these functions call the Base Function found in WorldInput.lua
+-- ===========================================================================
+
+-- ===========================================================================
+function UpdateDragMap()
+  if g_isMouseDragging then
+    -- Event sets a global in PlotInfo so tiles are not purchased while dragging
+    LuaEvents.CQUI_StartDragMap();
+  end
+
+  return BASE_CQUI_UpdateDragMap();
+end
+
+-- ===========================================================================
+function RealizeMovementPath(showQueuedPath:boolean)
+  print_debug("** Function Entry: RealizeMovementPath (CQUI Hook) showQueuedPath is: "..tostring(showQueuedPath));
+  if not UI.IsMovementPathOn() or UI.IsGameCoreBusy() then
+    return;
+  end
+    
+  -- CQUI (Azurency) : Check if in CITY_MANAGEMENT or STRIKE modes
+  local CQUI_im = UI.GetInterfaceMode();
+  if (CQUI_im == InterfaceModeTypes.CITY_MANAGEMENT or CQUI_im == InterfaceModeTypes.CITY_RANGE_ATTACK or CQUI_im == InterfaceModeTypes.DISTRICT_RANGE_ATTACK) then
+    return;
+  end
+
+  return BASE_CQUI_RealizeMovementPath( showQueuedPath );
+end
+
+-- ===========================================================================
+function OnUnitSelectionChanged( playerID:number, unitID:number, hexI:number, hexJ:number, hexK:number, isSelected:boolean, isEditable:boolean )
+  local msg = "** Function Entry: OnUnitSelectionChanged (CQUI Hook).  playerId:"..tostring(playerID).." unitId:"..tostring(unitID).." hexI:"..tostring(hexI).." hexJ:"..tostring(hexJ).." hexK:"..tostring(hexK).." isSelected:"..tostring(isSelected).." isEditable:"..tostring(isEditable);
+  print_debug(msg);
+  if playerID ~= Game.GetLocalPlayer() then
+    return;
+  end
+
+  -- CQUI (Azurency) : Fixes a Vanilla bug from SelectUnit.lua (not taking in account the district range attack)
+  -- CQUI (Azurency) : If a selection is occuring and the district attack interface mode is up, take it down.
+  if UI.GetInterfaceMode() == InterfaceModeTypes.DISTRICT_RANGE_ATTACK then
+    UI.SetInterfaceMode(InterfaceModeTypes.SELECTION);
+  end
+
+  return BASE_CQUI_OnUnitSelectionChanged( playerID, unitID, hexI, hexJ, hexK, isSelected, isEditable );
+end
+
+-- ===========================================================================
+function DefaultKeyDownHandler( uiKey:number )
+  print_debug("** Function Entry: DefaultKeyDownHandler (CQUI Hook).  uiKey: "..tostring(uiKey));
+  -- Note: This function always returns false, by design in Base game
+  BASE_CQUI_DefaultKeyDownHandler( uiKey ) ;
+
+  --CQUI Keybinds
+  local keyPanChanged :boolean = false;
+
+  if uiKey == Keys.VK_SHIFT then
+    print_debug("CQUI_isShiftDown = true");
+    CQUI_isShiftDown = true;
+  end
+
+  if uiKey == Keys.VK_ALT then
+    print_debug("CQUI_isAltDown = true");
+    CQUI_isAltDown = true;
+  end
+
+  if CQUI_hotkeyMode ~= CQUI_HOTKEYMODE_STANDARD then
```

Comment:
Seems unnecessary due to the condition below.

Reply:
fixed